### PR TITLE
Allow user to toggle between sorting by distance from residential and current location

### PR DESCRIPTION
**Pull request by narin**
## Description
If a user has opted to sort by distance from current location, they should be able to switch back to sorting by distance from their residential address and visa versa.

This PR replaces the `requestCurrentLocation()` action with a new `updateFacilitySortMethod()` action which accepts a sort method and will switch accordingly.  If we the user chooses to sort by distance from current location and we have not requested their coordinates yet, we also handle this within the action.

**Note:** The copy above the new toggle link is subject to change and will be handled in a different issue.

## Testing done
Local and unit.

## Screenshots
Screen Shot 2020-10-15 at 11 45 22 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx [hunk 1]
```diff
@@ -312,13 +337,13 @@ function mapStateToProps(state) {
...
-  updateFormData: actions.updateFormData,
+  checkEligibility: actions.checkEligibility,
   hideEligibilityModal: actions.hideEligibilityModal,
+  openFacilityPageV2: actions.openFacilityPageV2,
   routeToNextAppointmentPage: actions.routeToNextAppointmentPage,
   routeToPreviousAppointmentPage: actions.routeToPreviousAppointmentPage,
-  checkEligibility: actions.checkEligibility,
-  requestCurrentLocation: actions.requestCurrentLocation,
+  updateFacilitySortMethod: actions.updateFacilitySortMethod,
+  updateFormData: actions.updateFormData,
```
Comment (narin) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx:346]: Just sorting these alphabetically

File: src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx [hunk 2]
```diff
@@ -262,39 +286,40 @@ function VAFacilityPageV2({
...
+              <div aria-atomic="true" aria-live="assertive">
+                <AlertBox isVisible status="info" headline="Please wait">
+                  We’re checking if we can create an appointment for you at this
+                  facility. This may take up to a minute. Thank you for your
+                  patience.
+                </AlertBox>
+              </div>
+            )}
+          </SchemaForm>
+        )}
```
Comment (narin) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx:322]: Hide radio buttons while we determine user's location

File: src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx [hunk 3]
```diff
@@ -40,10 +40,10 @@ export default function FacilitiesRadioWidget({
         const checked = option.value === value;
         let distance;
 
-        if (sortMethod === FACILITY_SORT_METHODS.DISTANCE_FROM_RESIDENTIAL) {
+        if (sortMethod === FACILITY_SORT_METHODS.distanceFromResidential) {
           distance = legacyVAR?.distanceFromResidentialAddress;
         } else if (
-          sortMethod === FACILITY_SORT_METHODS.DISTANCE_FROM_CURRENT_LOCATION
+          sortMethod === FACILITY_SORT_METHODS.distanceFromCurrentLocation
```
Comment (narin) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/FacilitiesRadioWidget.jsx:46]: Renamed the objects within this constant to camel case

File: src/applications/vaos/utils/constants.js [hunk 4]
```diff
@@ -176,9 +176,9 @@ export const FACILITY_TYPES = {
 };
 
 export const FACILITY_SORT_METHODS = {
-  DISTANCE_FROM_RESIDENTIAL: 'distanceFromResidential',
-  DISTANCE_FROM_CURRENT_LOCATION: 'distanceFromCurrentLocation',
-  ALPHABETICAL: 'alphabetical',
+  distanceFromResidential: 'distanceFromResidentialAddress',
+  distanceFromCurrentLocation: 'distanceFromCurrentLocation',
+  alphabetical: 'alphabetical',
```
Comment (narin) [inline src/applications/vaos/utils/constants.js:181]: Renamed the objects within this constant to camel case to follow pattern of other constants

File: src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx [hunk 5]
```diff
@@ -253,6 +254,95 @@ describe('VAOS integration: VA flat facility page - multiple facilities', () =>
...
+    });
+    await setTypeOfCare(store, /primary care/i);
+
+    const screen = renderWithStoreAndRouter(<VAFacilityPage />, {
+      store,
+    });
+
+    await screen.findAllByRole('radio');
+    fireEvent.click(screen.getByText('use your current location'));
+    await screen.findByText('Finding your location...');
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx:295]: I would skip checking for the loading message in these tests. Based on the flaky tests in Jenkins, it doesn't seem like we can always rely on catching this text before it disappears.

This seems like a limitation of the fetch mocking approach we use, where we can't hold open a request and resolve it when we choose. I might put in a ticket to see if there's a way to build that functionality so we can reliably test loading states.

File: src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx [hunk 6]
```diff
@@ -253,6 +254,95 @@ describe('VAOS integration: VA flat facility page - multiple facilities', () =>
...
```
Comment (jbalboni) [inline src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx:342]: Same comment as above

File: src/applications/vaos/new-appointment/redux/actions.js [hunk 7]
```diff
@@ -427,24 +429,37 @@ export function openFacilityPageV2(page, uiSchema, schema) {
...
+        location = await getPreciseLocation();
+      } catch (e) {
+        captureError(e, false, 'facility page');
+        dispatch({
+          type: FORM_REQUEST_CURRENT_LOCATION_FAILED,
+        });
+      }
     }
+
+    dispatch({
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/redux/actions.js:457]: It seems sort of odd to me that this action fires regardless of whether or not the current location request succeeds. Wouldn't we not be changing the sort method in that case?

File: src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx (hunk 5 above)
Comment (narin) [reply src/applications/vaos/tests/new-appointment/components/VAFacilityPage/VAFacilityPageV2.multi.unit.spec.jsx]: Thanks for the heads up. Updated to look for radio buttons to reappear instead before checking for updated text.

File: src/applications/vaos/new-appointment/redux/actions.js (hunk 7 above)
Comment (narin) [reply src/applications/vaos/new-appointment/redux/actions.js]: Updated to only fire if location request was successful, or switching to other sort methods

File: src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx [hunk 8]
```diff
@@ -249,7 +270,10 @@ function VAFacilityPageV2({
             href="#"
```
[the remark this answers is not in the document]
Comment (narin) [reply src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx]: Replaced these with button links

File: src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx [hunk 9]
```diff
@@ -232,7 +232,10 @@ function VAFacilityPageV2({
                   href="#"
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/components/VAFacilityPage/VAFacilityPageV2.jsx:232]: This one should be a button, too

File: src/applications/vaos/new-appointment/redux/reducer.js [hunk 10]
```diff
@@ -439,12 +439,20 @@ export default function formReducer(state = initialState, action) {
...
-              b.legacyVAR.distancefromCurrentLocation,
-          );
+          });
+        }
+
+        requestLocationStatus = FETCH_STATUS.succeeded;
+      }
+
+      if (sortMethod === FACILITY_SORT_METHODS.alphabetical) {
+        typeOfCareFacilities = typeOfCareFacilities.sort(
```
Comment (jbalboni) [inline src/applications/vaos/new-appointment/redux/reducer.js:449]: Might want to make a note to try to write a test to cover this branch later.

**Review by jbalboni — APPROVED** [review]
Couple more minor comments, but looks good overall.

**Pull request merged**